src/algebraic_geometry/EllipticCurve.lean

```diff
@@ -10,56 +10,70 @@ import tactic.linear_combination
-We give a working definition of elliptic curves which is mathematically accurate
-in many cases, and also good for computation.
+We give a working definition of elliptic curves which is mathematically accurate in many cases,
+and also good for computation. The type of $K$-rational points on an elliptic curve over a field
+is defined as `EllipticCurve.point` in `src/algebraic_geometry/EllipticCurve/point.lean`.
 
 ## Mathematical background
 
-Let `S` be a scheme. The actual category of elliptic curves over `S` is a large category,
-whose objects are schemes `E` equipped with a map `E → S`, a section `S → E`, and some
-axioms (the map is smooth and proper and the fibres are geometrically connected group varieties
-of dimension one). In the special case where `S` is `Spec R` for some commutative ring `R`
-whose Picard group is trivial (this includes all fields, all principal ideal domains, and many
-other commutative rings) then it can be shown (using rather a lot of algebro-geometric machinery)
-that every elliptic curve is, up to isomorphism, a projective plane cubic defined by
-the equation `y² + a₁xy + a₃y = x³ + a₂x² + a₄x + a₆`, with `aᵢ : R`, and such that
-the discriminant of the aᵢ is a unit in `R`.
-
-Some more details of the construction can be found on pages 66-69 of
-[N. Katz and B. Mazur, *Arithmetic moduli of elliptic curves*][katz_mazur] or pages
-53-56 of [P. Deligne, *Courbes elliptiques: formulaire d'après J. Tate*][deligne_formulaire].
-
-## Warning
-
-The definition in this file makes sense for all commutative rings `R`, but it only gives
-a type which can be beefed up to a category which is equivalent to the category of elliptic
-curves over `Spec R` in the case that `R` has trivial Picard group `Pic R` or, slightly more
-generally, when its `12`-torsion is trivial. The issue is that for a general ring `R`, there
-might be elliptic curves over `Spec R` in the sense of algebraic geometry which are not
-globally defined by a cubic equation valid over the entire base.
-
-## TODO
-
-Define the `R`-points (or even `A`-points if `A` is an `R`-algebra). Care will be needed at infinity
-if `R` is not a field. Define the group law on the `R`-points. Prove associativity (hard).
+Let $S$ be a scheme. The actual category of elliptic curves over $S$ is a large category, whose
+objects are schemes $E$ equipped with a map $E \to S$, a section $S \to E$, and some axioms (the map
+is smooth and proper and the fibres are geometrically-connected one-dimensional group varieties). In
+the special case where $S = \mathrm{Spec}(R)$ for some commutative ring $R$ whose Picard group is
+trivial (this includes all fields, all PIDs, and many other commutative rings) it can be shown
+(using a lot of algebro-geometric machinery) that every elliptic curve is, up to isomorphism, a
+projective plane cubic defined by the equation $y^2 + a_1xy + a_3y = x^3 + a_2x^2 + a_4x + a_6$,
+with $a_i \in R$, and such that the discriminant of the $a_i$ is a unit in $R$.
```

Comment:
I don't think it's good idea to replace all ` ` and Unicode by LaTeX, especially given that the text is quite readable currently. It would make the page slower to load.

Reply:
The point is that backticks should be for actual Lean code whereas non-Lean mathematical code should be in LaTeX. I will revert things like $S$ and $E$ back to `S` and `E` but I'll keep the equation in LaTeX (since the actual Lean code is `E.a_i` etc).